src/app/loadout/ModPicker.tsx

```diff
@@ -39,191 +38,163 @@ const MAX_SLOT_INDEPENDENT_MODS = 5;
+
+        // Combat, general and raid mods are restricted across items so we need to manually
+        // set the max selectable
+        const maxSelectable = plugs.some((p) =>
+          slotSpecificPlugCategoryHashes.includes(p.plug.plugCategoryHash)
+        )
+          ? sockets.length
+          : MAX_SLOT_INDEPENDENT_MODS;
+
+        if (plugs.length && !plugSetsByHash[plugSetHash] && !(isArtificePlugSet && usedArtifice)) {
+          usedArtifice ||= isArtificePlugSet;
+          plugSetsByHash[plugSetHash] = {
+            plugSetHash,
+            maxSelectable,
+            selectionType: 'multi',
+            plugs,
+            selected: [],
+          };
+
+          // use an activity name, if one is available, for mods with no item type display name
+          if (!plugs[0].itemTypeDisplayName) {
+            const activityHash =
+              modMetadataByPlugCategoryHash[plugs[0].plug.plugCategoryHash]
+                ?.modGroupNameOverrideActivityHash;
+            if (activityHash) {
+              const activityName = defs.Activity.get(activityHash).displayProperties.name;
+              if (activityName) {
+                plugSetsByHash[plugSetHash].headerSuffix = activityName;
               }
             }
-          } else if (plugs.length && plugSetsByHash[plugSetHash]?.maxSelectable < sockets.length) {
-            plugSetsByHash[plugSetHash].maxSelectable = sockets.length;
           }
+        } else if (plugs.length && plugSetsByHash[plugSetHash]?.maxSelectable < sockets.length) {
+          plugSetsByHash[plugSetHash].maxSelectable = sockets.length;
         }
       }
+    }
 
-      const plugSets = Object.values(plugSetsByHash);
-
-      // Order our mods so that we assign the most picky mods first (the mods that have the fewest
-      // plugSets containing them). This is necessary for artifact mods in artificer sockets and
-      // essentially the same logic that actual mod assignment uses.
-      const orderedMods = _.sortBy(
-        lockedMods,
-        (mod) => plugSets.filter((s) => s.plugs.some((p) => p.hash === mod.resolvedMod.hash)).length
-      );
-
-      // However, sort the plugSets so that regular armor plugsets are preferred.
-      // Assigning the first artifact mod to the (Artifice Armor) header is confusing
-      // when a user doesn't have any artifice armor in the loadouts because the mod
-      // assignment might not actually need any artifice sockets.
-      plugSets.sort(compareBy((plugSet) => -plugSet.plugs.length));
+    const plugSets = Object.values(plugSetsByHash);
+
+    // However, sort the plugSets so that regular armor plugsets are preferred.
+    // Assigning the first artifact mod to the (Artifice Armor) header is confusing
+    // when a user doesn't have any artifice armor in the loadouts because the mod
+    // assignment might not actually need any artifice sockets.
+    plugSets.sort(compareBy((plugSet) => -plugSet.plugs.length));
```

Comment:
This part isn't needed right now but might as well keep it since it's not the only part of DIM's code that deals with these problems.